index.html

```diff
@@ -490,6 +490,157 @@ <h3 id="installation-sec">
+                member is the value of <code>eventInit.userChoice</code>.
+                </li>
+                <li>Resolve <a>[[\userResponsePromise]]</a>
+                <var>responseObj</var>.
+                </li>
+              </ol>
+            </li>
+          </ol>
+        </section>
+        <section>
+          <h4>
+            <code>prompt()</code> method
+          </h4>
+          <p>
+            The <dfn>prompt</dfn> method, when called, runs the following
+            steps:
+          </p>
+          <ol>
+            <li>Let <var>p</var> be a newly created promise.
+            </li>
+            <li>Resolve <var>p</var> with <a>[[\userResponsePromise]]</a>.
+            </li>
+            <li>If <a>[[\userResponsePromise]]</a> is pending:
+              <ol>
+                <li>If this event's <code>isTrusted</code> attribute is <code>
+                  false</code>, reject <a>[[\userResponsePromise]]</a> with
+                  <a>NotAllowedError</a>, optionally informing the developer
+                  that untrusted events can't call <code>prompt()</code>.
+                </li>
+                <li>Else if <a>[[\didPrompt]]</a> is <code>false</code>, then,
+                <a>in parallel</a>, <a>request to present an install prompt</a>
+                with this event. Wait, possibly indefinitely, for the end-user
+                to make a choice.
+                </li>
+              </ol>
+            </li>
+            <li>Return <var>p</var>.
+            </li>
+          </ol>
+          <p>
+            To <dfn>request to present an install prompt</dfn> with
+            <a>BeforeInstallPromptEvent</a> <var>event</var>:
+          </p>
+          <ol>
+            <li>
+              <a>Present an install prompt</a> and <var>outcome</var> be the
+              result.
+            </li>
+            <li>Let <var>responseObj</var> be a newly created
+            <a>PromptResponseObject</a> whose <code>userChoice</code> member is
+            the value of <var>outcome</var>.
+            </li>
+            <li>Resolve <a>[[\userResponsePromise]]</a> with
+            <var>responseObj</var>.
+            </li>
+          </ol>
+        </section>
+        <section class="informative">
+          <h4>
+            Usage example
+          </h4>
+          <p>
+            This example shows how one might prevent an automated install
+            prompt from showing until the user has finished a set of tasks.
+            Those tasks are represented as an array of promises, which the
+            application "awaits" to finish before an install prompt is
+            presented to the end-user.
+          </p>
+          <pre class="example" title="'beforeinstallprompt' in action">
+            window.addEventListener("beforeinstallprompt", async (event) =&gt; {
+              event.preventDefault();
+              // await e.g., user composing an email...
```

Comment:
This comment is a bit confusing. I think I understand it (if the user is composing an email in the application you would wait until they finish the task before prompting). But I think that's just a bit confusing of a use case and the primary use case for this is not to avoid interrupting the user, but to present in-context UI for prompting.

So how about:
```js
// Wait for e.g., the user to request installation from inside the app.
```